### ISSUE TITLE
"Provisional headers are shown" in devTools network panel when making request from dedicated web worker

### ISSUE THREAD

**Issue by tripl3inf** [nwjs/nw.js#7642]
NWJS Version : 0.50.2
Operating System : OSX 10.15.5 / Windows 7 Professional (Service Pack 1)

My team and I have recently started upgrading our NW application from version 0.36.1 to 0.50.2. One of the things we do is spin up a dedicated web worker from the background page on startup to offload our REST requests. The issue we are having is the devTools console is showing this "Provisional headers are shown" message on our requests, which blocks some relevant information that we need for debugging, etc. This only happens when making a request from the worker, not the background page or DOM side. We have done a lot of research into whats causing this, tried setting different headers/chrome args.. and none of it has worked. 

Expected behavior: 
See all request header information (cookies, etc) in devTools network tab without "Provisional headers are shown" message.

Actual behavior: 
"Provisional headers are shown" when making request from worker, blocking request header information.

![Screen Shot 2020-12-09 at 10 54 47 AM](https://user-images.githubusercontent.com/6551456/101665400-8cbdbf80-3a12-11eb-8fd1-3810436355da.png)

**Here is a simple test case to demonstrate:**

*package.json*
```json
{
  "name": "NWJS Test",
  "main": "starter.js",
  "chromium-args": "--mixed-context --enable-node-worker"
}
```

*starter.js*
```javascript
// open background page devTools
chrome.developerPrivate.openDevTools({
    renderViewId    : -1,
    renderProcessId : -1,
    extensionId     : chrome.runtime.id
});

// timeout just to let devTools open before making requests
setTimeout(() => {
    fetch('https://reqbin.com/echo/get/json')
        .then(response => response.json())
        .then(response => {
            console.log('background page response', response);
        });

    new Worker('TestWorker.js');
}, 6000)
```

*TestWorker.js*
```javascript
fetch('https://reqbin.com/echo/get/json')
    .then(response => response.json())
    .then(response => {
        console.log('worker response', response);
    });
```

You will notice that the request made from the background page behaves as expected, but the request made from the worker has the "Provisional headers are shown" message. Any insight into why this is happening would be greatly appreciated. Thanks in advance!

**Comment by rcruz00** [comment]
+1

**Comment by Irene125** [comment]
+1 saw same issue

**Comment by rogerwang** [comment]
In the sample, the 2nd request hits the cache, so there is no actual network request sent for  it. Thus no headers are shown.

**Comment by tripl3inf** [comment]
Based on your response, I have modified the test case. The request from TestWorker.js is now to a different endpoint that 
has NEVER been hit before:

*starter.js*
```javascript
// open background page devTools
chrome.developerPrivate.openDevTools({
    renderViewId    : -1,
    renderProcessId : -1,
    extensionId     : chrome.runtime.id
});

// timeout just to let devTools open before making requests
setTimeout(() => {
    fetch('https://reqbin.com/echo/get/json')
        .then(response => response.json())
        .then(response => {
            console.log('background page response', response);
        });

    new Worker('TestWorker.js');
}, 6000)
```
*TestWorker.js*
```javascript
fetch('https://jsonplaceholder.typicode.com/posts/3')
    .then(response => response.json())
    .then(response => {
        console.log('worker response', response);
    });
```

The problem persists, as the worker request still show provisional headers. 

You can see in this screenshot of devTools that the first request is now cached (size column show "(disk cache)", as we have made it before. The second request from the worker indicates that it is NOT from cache, as it has the size.
![devtools1](https://user-images.githubusercontent.com/6551456/107796928-f793bb80-6d1f-11eb-8703-07ddf8f767fb.png)

Looking at the details for the first request, you will see "Provisional headers are show. Disable cache to see full headers":
![devtools2](https://user-images.githubusercontent.com/6551456/107797036-18f4a780-6d20-11eb-8ae9-30eec8e28267.png)

Now looking at the details for the second request, we see a slightly different message, "Provisional headers are shown", but nothing about caching:
![devtools3](https://user-images.githubusercontent.com/6551456/107797505-a0dab180-6d20-11eb-89fa-126e27240886.png)

We have also run a proxy to confirm that the worker request is actually going out. 

Here is another test case, in which I have turned on the "Disable cache (while DevTools is open)" option in settings. I make the same request twice from the background page, and a third time from the worker:

*starter.js*
```javascript
setTimeout(() => {
    fetch('https://reqbin.com/echo/get/json')
        .then(response => response.json())
        .then(response => {
            console.log('background page response 1', response);

            fetch('https://reqbin.com/echo/get/json')
                .then(response => response.json())
                .then(response => {
                    console.log('background page response 2', response);

                    new Worker('TestWorker.js');
                });
        });

}, 3000)
```

*TestWorker.js*
```javascript
fetch('https://reqbin.com/echo/get/json')
    .then(response => response.json())
    .then(response => {
        console.log('worker response', response);
    });
```

Now, none of the requests show "(disk cache)" in the size column:
![devtools4](https://user-images.githubusercontent.com/6551456/107798557-ed72bc80-6d21-11eb-9a23-aede640d9554.png)

Details on the background page requests DO NOT show provisional headers warning, as expected:
![devtools5](https://user-images.githubusercontent.com/6551456/107798627-07ac9a80-6d22-11eb-991c-223bc8331ecd.png)

BUT, looking at the details for the request made from the worker, it still shows "Provisional headers are shown":
![devtools6](https://user-images.githubusercontent.com/6551456/107798834-480c1880-6d22-11eb-9306-c396f4e80dbd.png)

In conclusion, we are confident the worker request is not cached, yet the provisional headers message is still shown. We are unable to get accurate details on the headers when this message is present.

**Comment by tripl3inf** [comment]
And THANK YOU so much for your time. Not only in creating/maintaining NW, but also responding to my question.

**Comment by rogerwang** [comment]
This should be from the upstream change: `DevTools: do not expose raw headers for cross-origin requests https://chromium-review.googlesource.com/c/chromium/src/+/1313739`

Will look to fix it in NW soon.

Thanks for your patience waiting for the bug to be fixed. Your issue report makes the project better :)

**Comment by rogerwang** [comment]
Fixed in https://dl.nwjs.io/live-build/nw51/20210213-162000/87b949326/v0.51.2/

**Comment by tripl3inf** [comment]
AWESOME! THANKS AGAIN!
